src/Mapper.re

```diff
@@ -2,62 +2,73 @@ type address = int;
 type byte = int;
 
 type t = {
-  .
   get_prg: address => byte,
   set_prg: (address, byte) => unit,
   get_chr: address => byte,
   set_chr: (address, byte) => unit,
-  mirroring: Rom.mirroring,
+  mirroring: unit => Rom.mirroring,
   set_mirroring: Rom.mirroring => unit,
 };
 
-exception MapperNotImplemented(Rom.mapper);
+exception NotImplemented(Rom.mapper);
 exception NotAllowed(string);
 
-let aref = (mem, address) => Bytes.get(mem, address) |> Char.code;
-let set = (mem, address, byte) => Char.chr(byte) |> Bytes.set(mem, address);
-
 let nrom = (rom: Rom.t): t => {
-  val mirroring = ref(rom.mirroring);
-  pub get_prg = address => aref(rom.prg, address land (rom.prg_size - 1));
-  pub set_prg = (_, _) => raise(NotAllowed("Cannot write to prg"));
-  pub get_chr = address => aref(rom.chr, address);
-  pub set_chr = (address, byte) => set(rom.chr, address, byte);
-  pub mirroring = mirroring^;
-  pub set_mirroring = style => mirroring := style
+  let mirroring = ref(rom.mirroring);
+
+  {
+    get_prg: address => Util.aref(rom.prg, address land (rom.prg_size - 1)),
+    set_prg: (_, _) => raise(NotAllowed("Cannot write to prg")),
+    get_chr: address => Util.aref(rom.chr, address),
+    set_chr: (address, byte) => Util.set(rom.chr, address, byte),
+    mirroring: () => mirroring^,
+    set_mirroring: style => mirroring := style,
```

Comment:
Something seems not quite right about mirroring. A couple things:

* I've had to add new mirroring types for MMC1. Should those be globally visible?
* All of these mappers duplicate `mirroring` and `set_mirroring`. Is this something that should be lifted to the PPU?
* Why do we have a `set_mirroring` at all? It seems like we'd be setting mirroring via an appropriate PRG write.